src/lambda.ts

```diff
@@ -0,0 +1,41 @@
+import { configure as serverlessExpress } from '@vendia/serverless-express';
+import { NestFactory } from '@nestjs/core';
+import { AppModule } from './app.module';
+import { SwaggerModule, DocumentBuilder } from '@nestjs/swagger';
+import { MigrateDeploy } from '@prisma/migrate/dist/commands/MigrateDeploy';
+
+let cachedServer;
+let shouldMigrate = true;
+
+export const handler = async (event, context) => {
+  // It is very annoying to run prisma migrations programatically...
+  //
+  // Hack adapted from:
+  //  https://github.com/prisma/prisma/issues/4703#issuecomment-1430112147
+  if (shouldMigrate) {
+    await new MigrateDeploy().parse([]);
+    shouldMigrate = false;
+  }
```

Comment:
Conheço pouco sobre essa framework prisma, mas penso que essa parte de migração poderia executar apenas no deploy. Pois assim como está, entendo que executa a cada invocação da Lambda, ou seja, a cada requisição recebida. 
Para isso teriamos que separar esse trecho em um script separado e chamar ele no deploy:
```
// migrate.ts

const { MigrateDeploy } = require('@prisma/migrate/dist/commands/MigrateDeploy');

export const handler = async () => {
  await new MigrateDeploy().parse([]);
};
```

Reply:
@lucasfacchini o ciclo de vida das variáveis globais é associado a cada instância de uma funcão, não a cada invocação. 

Então, pelo que entendi, não executaria a cada invocação, e sim toda vez que uma nova instancia for criada (por, exemplo quando houver um cold start, ou chegar uma nova requisicão e todas as instâncias já estiverem ocupadas) 

O mesmo vale pra inicialização do `cachedServer` abaixo. É isso mesmo, @meyer1994 ?

Reply:
Verdade, caso a instância for reutilizada as variáveis globais manteriam os mesmos valores. Nesse caso vai depender de como o serviço Lambda gerencia essa questão. 
Acho que mitiga a preocupação inicial, mas em algum momento futuro eu separaria esse trecho e executaria apenas no deploy.